binary distribution of mc for arm64 (ARMv8, aarch64)?

## Expected behaviour

https://dl.minio.io/client/mc/release/ would have downloadable arm64 (aarch64) binaries.

## Actual behaviour

Binaries not found.

## Steps to reproduce the behaviour

1. visit https://dl.minio.io/client/mc/release/
2. inspect directory

## mc version
- (paste output of `mc version`)

## System information

To be installed on Packet 2A, 96-core ARMv8 server.

When I try to build from source I get this, fyi:

```
root@docker-build-test:/mnt/src/minio/mc# make checks
Checking deps:
Check for supported arch.. Invalid arch: Linux aarch64 not supported
Makefile:6: recipe for target 'checks' failed
make: *** [checks] Error 1
```

The relevant bit of output is
```
$ arch
aarch64
```

--- a/buildscripts/build.sh
+++ b/buildscripts/build.sh
@@ -23,7 +23,7 @@ _init() {
     fi
 
     # List of supported architectures
-    SUPPORTED_OSARCH='linux/386 linux/amd64 linux/arm windows/386 windows/amd64 freebsd/amd64'
+    SUPPORTED_OSARCH='linux/386 linux/amd64 linux/arm linux/arm64 windows/386 windows/amd64 freebsd/amd64'
 
     ## System binaries
     CP=`which cp`
@@ -46,19 +46,55 @@ go_build() {
     # Release shasum name
     release_shasum="$release_str/$os-$arch/$(basename $package).shasum"
 
-    # Using `go build` to build the binary.
-    CGO_ENABLED=0 GOOS=$os GOARCH=$arch go build --ldflags "${LDFLAGS}" -o $release_bin
+  # Go build to build the binary.
+    if [ "${arch}" == "arm" ]; then
+        # Release binary downloadable name
+        release_real_bin_6="$release_str/$os-${arch}6vl/$(basename $package)"
 
-    # Create copy
-    if [ $os == "windows" ]; then
-        $CP -p $release_bin ${release_real_bin}.exe
+        release_bin_6="$release_str/$os-${arch}6vl/$(basename $package).$release_tag"
+        ## Support building for ARM6vl
+        CGO_ENABLED=0 GOARM=6 GOOS=$os GOARCH=$arch go build --ldflags "${LDFLAGS}" -o $release_bin_6
+
+        ## Copy
+        $CP -p $release_bin_6 $release_real_bin_6
+
+        # Release shasum name
+        release_shasum_6="$release_str/$os-${arch}6vl/$(basename $package).shasum"
+
+        # Calculate shasum
+        shasum_str=$(${SHASUM} ${release_bin_6})
+        echo ${shasum_str} | $SED "s/$release_str\/$os-${arch}6vl\///g" > $release_shasum_6
+
+        # Release binary downloadable name
+        release_real_bin_7="$release_str/$os-$arch/$(basename $package)"
+
+        release_bin_7="$release_str/$os-$arch/$(basename $package).$release_tag"
+        ## Support building for ARM7vl
+        CGO_ENABLED=0 GOARM=7 GOOS=$os GOARCH=$arch go build --ldflags "${LDFLAGS}" -o $release_bin_7
+
+        ## Copy
+        $CP -p $release_bin_7 $release_real_bin_7
+
+        # Release shasum name
+        release_shasum_7="$release_str/$os-$arch/$(basename $package).shasum"
+
+        # Calculate shasum
+        shasum_str=$(${SHASUM} ${release_bin_7})
+        echo ${shasum_str} | $SED "s/$release_str\/$os-$arch\///g" > $release_shasum_7
     else
-        $CP -p $release_bin $release_real_bin
-    fi
+        CGO_ENABLED=0 GOOS=$os GOARCH=$arch go build --ldflags "${LDFLAGS}" -o $release_bin
 
-    # Calculate shasum
-    shasum_str=$(${SHASUM} ${release_bin})
-    echo ${shasum_str} | $SED "s/$release_str\/$os-$arch\///g" > $release_shasum
+        # Create copy
+        if [ $os == "windows" ]; then
+            $CP -p $release_bin ${release_real_bin}.exe
+        else
+            $CP -p $release_bin $release_real_bin
+        fi
+
+        # Calculate shasum
+        shasum_str=$(${SHASUM} ${release_bin})
+        echo ${shasum_str} | $SED "s/$release_str\/$os-$arch\///g" > $release_shasum
+    fi
 }
 
 main() {

--- a/buildscripts/checkdeps.sh
+++ b/buildscripts/checkdeps.sh
@@ -23,7 +23,8 @@ _init() {
     GIT_VERSION="1.0"
     GO_VERSION="1.7.1"
     OSX_VERSION="10.8"
-    UNAME=$(uname -sm)
+    KNAME=$(uname -s)
+    ARCH=$(uname -m)
 
     ## Check all dependencies are present
     MISSING=""
@@ -145,54 +146,40 @@ check_golang_env() {
     fi
 }
 
-is_supported_os() {
-    local supported
-    case ${UNAME%% *} in
-        "Linux")
-	    supported=1
+assert_is_supported_os() {
+    case "${KNAME}" in
+        Linux | FreeBSD )
+            return
             ;;
-        "Darwin")
+        Darwin )
             osx_host_version=$(env sw_vers -productVersion)
-            check_version "${osx_host_version}" "${OSX_VERSION}"
-            [[ $? -ge 2 ]] && die "Minimum OSX version supported is ${OSX_VERSION}"
-	    supported=1
-            ;;
-        "SunOS")
-	    supported=1
-            ;;
-        "FreeBSD")
-            supported=1
-	    ;;
-        "*")
-	    supported=0
-	    ;;
-    esac
-    if [ "x$supported" != "x1" ]; then
-        echo "Invalid os: ${UNAME} not supported"
-        exit 1;
-    fi
-}
-
-is_supported_arch() {
-    local supported
-    case ${UNAME##* } in
-        "x86_64" | "amd64")
-            supported=1
-            ;;
-        "arm"*)
-            supported=1
-            ;;
-        "i86pc"*)
-            supported=1
+            if ! check_minimum_version "${OSX_VERSION}" "${osx_host_version}"; then
+                echo "ERROR"
+                echo "OSX version '${osx_host_version}' not supported."
+                echo "Minimum supported version: ${OSX_VERSION}"
+                exit 1
+            fi
+            return
             ;;
         *)
-            supported=0
-            ;;
+            echo "ERROR"
+            echo "OS '${KNAME}' is not supported."
+            echo "Supported OS: [Linux, FreeBSD, Darwin]"
+            exit 1
     esac
-    if [ "x$supported" != "x1" ]; then
-        echo "Invalid arch: ${UNAME} not supported"
-        exit 1;
-    fi
+}
+
+assert_is_supported_arch() {
+    case "${ARCH}" in
+        x86_64 | amd64 | aarch64 | arm* )
+            return
+            ;;
+        *)
+            echo "ERROR"
+            echo "Arch '${ARCH}' not supported."
+            echo "Supported Arch: [x86_64, amd64, aarch64, arm*]"
+            exit 1
+    esac
 }
 
 check_deps() {
@@ -209,10 +196,10 @@ check_deps() {
 
 main() {
     echo -n "Check for supported arch.. "
-    is_supported_arch
+    assert_is_supported_arch
 
     echo -n "Check for supported os.. "
-    is_supported_os
+    assert_is_supported_os
 
     echo -n "Checking if proper environment variables are set.. "
     check_golang_env

